presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/OptimizeMixedDistinctAggregations.java

```diff
@@ -480,19 +517,19 @@ private static IfExpression createIfExpression(Expression left, Expression right
     private static class AggregateInfo
     {
         private final List<Symbol> groupBySymbols;
-        private final Symbol mask;
+        private final List<Symbol> masks;
```

Comment:
I think if you change this to `Map<Symbol, Symbol>` (key is the same as `aggregations`, value is the mask symbol) it would make the logic easier to reason about.